examples/simple/main.go

```diff
@@ -0,0 +1,130 @@
+
+	// Create a New Relic OpenTelemetry Exporter
+	apiKey, ok := os.LookupEnv("NEW_RELIC_API_KEY")
+	if !ok {
+		fmt.Println("missing NEW_RELIC_API_KEY required for New Relic OpenTelemetry Exporter")
+	}
+
+	exporter, err := newrelic.NewExporter(
+		"Sample OpenTelemetry Service",
+		apiKey,
+		telemetry.ConfigBasicErrorLogger(os.Stderr),
+		telemetry.ConfigBasicDebugLogger(os.Stderr),
+		telemetry.ConfigBasicAuditLogger(os.Stderr),
+	)
+	if err != nil {
+		fmt.Printf("failed to instantiate New Relic OpenTelemetry exporter: %v\n", err)
+	}
+
+	// Create a tracer provider
+	ctx := context.Background()
+	bsp := sdktrace.NewBatchSpanProcessor(exporter)
+	tp := sdktrace.NewTracerProvider(sdktrace.WithSpanProcessor(bsp))
+	defer func() { _ = tp.Shutdown(ctx) }()
+
+	// Create a meter provider
+	pusher := push.New(
+		basic.New(
+			simple.NewWithExactDistribution(),
+			exporter,
+		),
+		exporter,
+	)
+	pusher.Start()
+	defer pusher.Stop()
+
+	// Set global options
+	otel.SetTracerProvider(tp)
+	otel.SetMeterProvider(pusher.MeterProvider())
+	propagator := propagation.NewCompositeTextMapPropagator(propagation.Baggage{}, propagation.TraceContext{})
+	otel.SetTextMapPropagator(propagator)
+
+	// Sample metric instruments
+	fooKey := label.Key("ex.com/foo")
+	barKey := label.Key("ex.com/bar")
+	lemonsKey := label.Key("ex.com/lemons")
+	anotherKey := label.Key("ex.com/another")
+
+	commonLabels := []label.KeyValue{lemonsKey.Int(10), label.String("A", "1"), label.String("B", "2"), label.String("C", "3")}
+
+	meter := otel.Meter("ex.com/basic")
+
+	observerCallback := func(_ context.Context, result metric.Float64ObserverResult) {
+		result.Observe(1, commonLabels...)
+	}
+	_ = metric.Must(meter).NewFloat64ValueObserver("ex.com.one", observerCallback,
+		metric.WithDescription("A ValueObserver set to 1.0"),
+	)
+
+	valueRecorder := metric.Must(meter).NewFloat64ValueRecorder("ex.com.two")
+
+	boundRecorder := valueRecorder.Bind(commonLabels...)
+	defer boundRecorder.Unbind()
+
+	// Create a trace and some measurements
+	tracer := otel.Tracer("ex.com/basic")
+	ctx = baggage.ContextWithValues(ctx,
+		fooKey.String("foo1"),
+		barKey.String("bar1"),
+	)
+
+	func(ctx context.Context) {
+		var span trace.Span
+		ctx, span = tracer.Start(ctx, "operation")
+		defer span.End()
+
+		span.AddEvent("Nice operation!", trace.WithAttributes(label.Int("bogons", 100)))
+		span.SetAttributes(anotherKey.String("yes"))
+
+		meter.RecordBatch(
+			// Note: call-site variables added as context Entries:
+			baggage.ContextWithValues(ctx, anotherKey.String("xyz")),
+			commonLabels,
+
+			valueRecorder.Measurement(2.0),
+		)
+
+		func(ctx context.Context) {
+			var span trace.Span
+			ctx, span = tracer.Start(ctx, "Sub operation...")
+			defer span.End()
+
+			span.SetAttributes(lemonsKey.String("five"))
+			span.AddEvent("Sub span event")
+			boundRecorder.Record(ctx, 1.3)
+		}(ctx)
+	}(ctx)
+
+	// Wait for the New Relic OpenTelemetry Exporter to send data to New Relic
```

Comment:
I'd suggest replacing this sleep with `defer exporter.Shutdown(ctx)` right after successful creation of the exporter. That way, when this function returns, the exporter should flush its buffers.

Reply:
Much better, thanks.